### PR TITLE
add support for portal links

### PR DESCRIPTION
**Pull request by Dotunj**

### CODE REVIEW

File: datastore/mongo/mongo.go [hunk 1]
```diff
@@ -410,6 +411,17 @@ func compoundIndices() map[string][]mongo.IndexModel {
 				Options: options.Index().SetUnique(true),
 			},
 		},
+
+		datastore.PortalLinkCollection: {
+			{
+				Keys: bson.D{
+					{Key: "token", Value: 1},
+					{Key: "deleted_at", Value: 1},
+					{Key: "document_status", Value: 1},
```
Comment (jirevwe) [inline datastore/mongo/mongo.go:420]: we can remove this since it's not going to be used anymore

File: datastore/models.go [hunk 2]
```diff
@@ -832,6 +833,21 @@ type OrganisationInvite struct {
...
+type PortalLink struct {
+	ID                primitive.ObjectID `json:"-" bson:"_id"`
+	UID               string             `json:"uid" bson:"uid"`
+	Name              string             `json:"name" bson:"name"`
+	GroupID           string             `json:"group_id" bson:"group_id"`
+	Token             string             `json:"-" bson:"token"`
+	Endpoints         []string           `json:"endpoints" bson:"endpoints"`
+	EndpointsMetadata []Endpoint         `json:"endpoints_metadata" bson:"endpoints_metadata"`
+
+	DocumentStatus DocumentStatus     `json:"-" bson:"document_status"`
```
Comment (jirevwe) [inline datastore/models.go:845]: here too

**Review by jirevwe — APPROVED** [review]

File: datastore/models.go (hunk 2 above)
Comment (Subomi) [inline datastore/models.go:845]: @Dotunj We no longer use `DocumentStatus`

File: datastore/mongo/portal_link.go [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+	}
+
+	if !util.IsStringEmpty(f.EndpointID) {
+		filter["endpoints"] = f.EndpointID
+
+		matchStage = bson.D{
+			{Key: "$match",
+				Value: bson.D{
+					{Key: "group_id", Value: groupID},
+					{Key: "endpoints", Value: f.EndpointID},
```
Comment (Subomi) [inline datastore/mongo/portal_link.go:99]: I am not sure but will this work, given `endpoints` is an array?

Comment (Dotunj) [reply datastore/mongo/portal_link.go:99]: Yeah it does work. See [here](https://www.mongodb.com/docs/manual/tutorial/query-arrays/#query-an-array-for-an-element)

**Review by Subomi — APPROVED** [review]

**Pull request merged**